subsys/net/lib/sntp/sntp.c

```diff
@@ -116,7 +117,7 @@ static int sntp_recv_response(struct sntp_ctx *sntp, uint32_t timeout,
 		return -ETIMEDOUT;
 	}
 
-	rcvd = zsock_recv(sntp->sock.fd, (uint8_t *)&buf, sizeof(buf), 0);
+	rcvd = recv(sntp->sock.fd, (uint8_t *)&buf, sizeof(buf), 0);
```

Comment:
this should probably be unchanged

Reply:
It's using the `zsock_` version the prefered way?

If I try to change it back the compiler doesn't know what `zsock_recv` is for example, and they all fail:
```
/home/hein/zephyrproject/zephyr/subsys/net/lib/sntp/sntp.c:111:18: warning: implicit declaration of function ‘zsock_poll’; did you mean ‘k_poll’? [-Wimplicit-function-declaration]
  111 |         status = zsock_poll(sntp->sock.fds, sntp->sock.nfds, timeout);
      |                  ^~~~~~~~~~
      |                  k_poll
```
Which means I would have to `#include <zephyr/net/socket.h>`, but that clashes with `#include <sys/socket.h>` in `sntp.h` and creates many redeclaration errors:
```
In file included from /usr/include/sys/socket.h:26,
                 from /usr/include/netinet/in.h:23,
                 from /usr/include/arpa/inet.h:22,
                 from /home/hein/zephyrproject/zephyr/include/zephyr/net/sntp.h:12,
                 from /home/hein/zephyrproject/zephyr/subsys/net/lib/sntp/sntp.c:12:
/usr/include/bits/types/struct_iovec.h:26:8: error: redefinition of ‘struct iovec’
   26 | struct iovec
      |        ^~~~~
In file included from /home/hein/zephyrproject/zephyr/include/zephyr/net/socket.h:27,
                 from /home/hein/zephyrproject/zephyr/subsys/net/lib/sntp/sntp.c:11:
/home/hein/zephyrproject/zephyr/include/zephyr/net/net_ip.h:229:8: note: originally defined here
  229 | struct iovec {
      |        ^~~~~
```
And I I replace `#include <sys/socket.h>` with `#include <zephyr/net/socket.h>` (and remove `<poll>`) it also doesn't work, and results in a new cascade of errors:
```
In file included from /usr/include/arpa/inet.h:22,
                 from /home/hein/zephyrproject/zephyr/include/zephyr/net/sntp.h:12,
                 from /home/hein/zephyrproject/zephyr/subsys/net/lib/sntp/sntp_simple.c:9:
/home/hein/zephyrproject/zephyr/include/zephyr/net/net_ip.h:63:9: error: redeclaration of enumerator ‘IPPROTO_IP’
   63 |         IPPROTO_IP = 0,            /**< IP protocol (pseudo-val for setsockopt() */
      |         ^~~~~~~~~~
/usr/include/netinet/in.h:42:5: note: previous definition of ‘IPPROTO_IP’ with type ‘enum <anonymous>’
   42 |     IPPROTO_IP = 0,        /* Dummy protocol for TCP.  */
      |     ^~~~~~~~~~
```
I played around with changing more includes, but I always run into redeclaration errors or implicit declarations. I'm also not sure which is the right way to go about it. The only way I could get it to compile was doing this change. But please let me know if there is a better way :)